### PR TITLE
Deprecate fields

### PR DESCRIPTION
**Pull request by dbenavidesv**
## Summary
- Deprecated `lastName` field, now the full name of the user will be displayed in `name` field
- Added type resolver for `name` field to get the full name
- Refactored queries and mutations, added resolver and service methods
- Modified tests to comply with new field structure

## Trello Card
https://trello.com/c/qPb1eJgG

### CODE REVIEW

File: app/graphql/users/resolvers.js [hunk 1]
```diff
@@ -16,3 +17,26 @@ exports.createUser = (root, { user }) => {
       throw error;
     });
 };
+
+exports.getUser = (root, args) => {
+  logger.info(`Fetching user with ${JSON.stringify(args)}`);
```
Comment (gonziesc) [inline app/graphql/users/resolvers.js]: use util inspect

File: app/graphql/users/queries.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   queries: {
-    user: (_, params) => User.getOne(params),
-    users: (_, params) => User.getAll(params)
+    user: resolvers.getUser,
+    users: resolvers.getUsers
   },
   schema: gql`
     extend type Query {
-      user(id: ID, firstName: String, email: String): User!
+      user(id: ID, name: String, email: String): User!
```
Comment (gonziesc) [inline app/graphql/users/queries.js:12]: use deprecation policy

File: app/graphql/users/resolvers.js (hunk 1 above)
Comment (dbenavidesv) [reply app/graphql/users/resolvers.js]: Got it

File: app/graphql/users/queries.js (hunk 2 above)
Comment (dbenavidesv) [reply app/graphql/users/queries.js:12]: Got it

Comment (dbenavidesv) [reply app/graphql/users/queries.js:12]: I'll do it in field definition since doing it in argument definition has no effect

**Review by gonziesc — APPROVED** [review]

**Pull request merged**